ui/components/toolbar.go

```diff
@@ -13,14 +14,31 @@ func NewToolBar(
 	pullRequestMenuItemClickedEventHandler critic.PullRequestMenuItemClickedEventHandler,
 	analyzeButtonHandler critic.AnalyzeButtonClickedEventHandler,
 ) fyne.CanvasObject {
+	selectedGPTModel := ""
+	gptModelSelect := widget.NewSelect(critic.SupportedGPTModels, func(selected string) {
+		selectedGPTModel = selected
+	})
+
+	// Wrapper function to check GPT model selection before executing the handler
+	executeWithModelCheck := func(handler func(string)) func() {
```

Comment:
Maybe we should inject this as a callback like
https://github.com/kmesiab/ai-code-critic/pull/46/files#diff-bd4412fcb2e3485d51889e0ea7a885fe2a523ae6e95363be918c5fac35aede9eR14

Reply:
this was my initial idea, however, I found it very difficult to keep track of the `selectedGPTModel` field across handlers, callbacks, etc.

Reply:
Fair.  

Do you think it would make sense to add it here: https://github.com/kmesiab/ai-code-critic/blob/main/ui/critic_window.go ?  

What about writing it to the config?

Reply:
Adding to the critic_window I believe would make sense...
config I think should strictly be for application configuration
another route we can take is having the model be selected on the pull request modal instead of toolbar, then on press of analyze button we can take gpt modal value from the critic `window.PullRequestURLModal.GPTModel`, I have provided example screenshots 
![image](https://github.com/kmesiab/ai-code-critic/assets/34355192/a22532e3-e13e-49d9-bc2c-d9aed795a026)
![image](https://github.com/kmesiab/ai-code-critic/assets/34355192/0663d1fa-0843-4276-b4e0-4fd4242c03aa)
since PR form must be submitted before analyze button press this should always work, WDYT?